### ISSUE TITLE
No voices, no main menu audio

### ISSUE THREAD

**Issue by HerrTiSo** [krupar101/f4london_steam_deck_ini#1]
Somehow, your .ini results in no main menu audio, no voices being played, and issues like that.

PS: We're temporarily utilizing your optimized .ini for an automation (kind of one-click) we've built to set up FOLON on Steam Deck. I'll share the link once it goes live.

**Comment by krupar101** [comment]
Hi there. 
The audio issues are likely not related to the .INI file but to missing FAudio.dll in the wine prefix. This is only relevant for the GoG version as Steam seems to install basic libraries during the initial prefix setup after installing Fallout 4. 

You can also use this script as reference: 
https://github.com/krupar101/f4london_steam_deck_ini/blob/main/verify_f4london_steam_deck_installation.sh

I created it today to help people troubleshoot the installation. Also added a post about it: 
https://www.reddit.com/r/fallout4london/comments/1ebrc74/comment/lfmv0e2/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

Potentially you can use it to verify if your script installed everything correctly :) 

One more remark. The Fallout 4.ini file shipped with the mod on GoG seems to have a missing line - because of this the game did not want to start for me. 

The missing line is:
sResourceDataDirsFinal=

It should be added as last line in the [Archive] section of the file. You can see this in my .INI file.

The second problem with the one that shipped with the mod is that the .ini is lowercase. 
During my testing f4se_loader.exe expects the extension to be uppercase.

**Comment by HerrTiSo** [comment]
> You can also use this script as reference: https://github.com/krupar101/f4london_steam_deck_ini/blob/main/verify_f4london_steam_deck_installation.sh
> 
> I created it today to help people troubleshoot the installation. Also added a post about it: https://www.reddit.com/r/fallout4london/comments/1ebrc74/comment/lfmv0e2/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button
> 
> Potentially you can use it to verify if your script installed everything correctly :)

it states "Configuration file for 149... not found." which might stem from the fact we're using NonSteamLaunchers instead of Heroic?

**Comment by krupar101** [comment]
> > You can also use this script as reference: https://github.com/krupar101/f4london_steam_deck_ini/blob/main/verify_f4london_steam_deck_installation.sh
> > I created it today to help people troubleshoot the installation. Also added a post about it: https://www.reddit.com/r/fallout4london/comments/1ebrc74/comment/lfmv0e2/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button
> > Potentially you can use it to verify if your script installed everything correctly :)
> 
> it states "Configuration file for 149... not found." which might stem from the fact we're using NonSteamLaunchers instead of Heroic?

That is very likely as most of the files for GoG installation are checked in $HOME/Games/Heroic since this is the default installation path. 

I could adjust it for you but I would need to know 2 paths: 
1. Fallout 4 Game Files location
2. Wine Prefix location 

I suspect only 1 wine prefix will be used for Fallout 4 and Fallout London - Which is a bit unlike Heroic Laucher :)

**Comment by krupar101** [comment]
> > You can also use this script as reference: https://github.com/krupar101/f4london_steam_deck_ini/blob/main/verify_f4london_steam_deck_installation.sh
> > I created it today to help people troubleshoot the installation. Also added a post about it: https://www.reddit.com/r/fallout4london/comments/1ebrc74/comment/lfmv0e2/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button
> > Potentially you can use it to verify if your script installed everything correctly :)
> 
> it states "Configuration file for 149... not found." which might stem from the fact we're using NonSteamLaunchers instead of Heroic?

Can't wait for your feedback on the new folon installer. 
https://overkill.wtf/p/1a37b2b3-bded-4628-aadf-4e6f5240d4fe/

I cooperated with Kevin to improve the process and to make it as seamless as possible. :)

**Comment by HerrTiSo** [comment]
> > > You can also use this script as reference: https://github.com/krupar101/f4london_steam_deck_ini/blob/main/verify_f4london_steam_deck_installation.sh
> > > I created it today to help people troubleshoot the installation. Also added a post about it: https://www.reddit.com/r/fallout4london/comments/1ebrc74/comment/lfmv0e2/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button
> > > Potentially you can use it to verify if your script installed everything correctly :)
> > 
> > 
> > it states "Configuration file for 149... not found." which might stem from the fact we're using NonSteamLaunchers instead of Heroic?
> 
> Can't wait for your feedback on the new folon installer. https://overkill.wtf/p/1a37b2b3-bded-4628-aadf-4e6f5240d4fe/
> 
> I cooperated with Kevin to improve the process and to make it as seamless as possible. :)

Yeah, he told me about this already, and I'm pumped to try it out tonight!!!

**Comment by HerrTiSo** [comment]
<img width="661" alt="image" src="https://github.com/user-attachments/assets/18977cee-9fe7-4200-896c-e6581575e498"> :(

**Comment by krupar101** [comment]
use this: 
curl -O https://raw.githubusercontent.com/overkillwtf/folon-steamdeck-installer/main/fallout.sh && chmod +x fallout.sh && ./fallout.sh

Pretty sure you are trying to run an html page as a script :D

**Comment by HerrTiSo** [comment]
Right. But somehow, in the end, the FOLON installer wasn't able to see FO4 under D:
![Screenshot_20240807_004037](https://github.com/user-attachments/assets/1b9c39d8-688a-4089-b423-557f788ce298)

**Comment by HerrTiSo** [comment]
Okay, so - I was able to circumvent this symlink issue by picking the location myself. Everything worked out, besides a weird bug where my onscreen keyboard simply didn't want to live anymore and I had to reboot the Deck. But lo and behold, the failsafe mechanism which offers to proceed at the recent step of the script worked.

**Now two major considerations:**
- How are we going to deal with the upcoming FOLON updates, especially the one which makes the dedicated Buffet mod unnecessary?
- Isn't your Steam Deck optimized .ini applied? Because the game resorts to 16:9 now.

Bottom line - tremendous work!

**Comment by krupar101** [comment]
> Okay, so - I was able to circumvent this symlink issue by picking the location myself. Everything worked out, besides a weird bug where my onscreen keyboard simply didn't want to live anymore and I had to reboot the Deck. But lo and behold, the failsafe mechanism which offers to proceed at the recent step of the script worked.
> 
> **Now two major considerations:**
> 
> * How are we going to deal with the upcoming FOLON updates, especially the one which makes the dedicated Buffet mod unnecessary?
> * Isn't your Steam Deck optimized .ini applied? Because the game resorts to 16:9 now.
> 
> Bottom line - tremendous work!

Good questions :)

- If FOLON team create an update Heroic Launcher will show that info and is capable of updating the installer files. Once the updated files are installed by heroic we can add an option when running the script from the beginning to either install the game or update the game. Install will start from beginning and update from step 7. (the Buffout 4 step firstly checks if the mod is installed and only proceeds to install it if it's not recognized. If they include it the installation step should just be skipped.)
- My .ini file does not contain the information about the screen size. This information is stored in Fallout4Prefs.ini. Currently our installer removes this file because it caused issues on some steam decks and prevented the game from launching. The game creates this file automatically during first run and defaults to 16:9 - that is why you ended up with black bars and 16:9 resolution.
- I was trying to debug the symlink issue. To be honest I am not sure what causes it. The symlink sometimes gets correctly created and sometimes doesn't. Looks like often Proton removes it whenever it starts the GoG Installer. I will try to find a solution for that but for now that causes some problems.

**Comment by krupar101** [comment]
closing the issue.
Topic is solved
